morpheus/stages/inference/inference_stage.py

```diff
@@ -299,8 +299,10 @@ def set_output_fut(resp: TensorMemory, inner_batch, batch_future: mrc.Future):
         else:
             node = builder.make_node(self.unique_name, ops.build(py_inference_fn))
 
-        # Set the concurrency level to be up with the thread count
-        node.launch_options.pe_count = self._thread_count
+            # Set the concurrency level to be up with the thread count, intentionally only setting this for python
+            # implementations to avoid #1639
+            node.launch_options.pe_count = self._thread_count
+
```

Comment:
I have confirmed that this has nothing to do with the inference stage itself. Its just the Triton inference stage. Lets move this into the derived class `TritonInferenceStage`.